applications/convection_diffusion_application/custom_elements/laplacian_element.cpp

```diff
@@ -92,12 +101,14 @@ void LaplacianElement::CalculateLocalSystem(MatrixType& rLeftHandSideMatrix, Vec
 
         //Calculating the cartesian derivatives (it is avoided storing them to minimize storage)
         noalias(DN_DX) = prod(DN_De[PointNumber],InvJ0);
-
-        double IntToReferenceWeight = integration_points[PointNumber].Weight() * DetJ0;
+        
+        const double IntToReferenceWeight = integration_points[PointNumber].Weight() * DetJ0;
         noalias(rLeftHandSideMatrix) += IntToReferenceWeight * prod(DN_DX, trans(DN_DX)); //
+
+        // Calculating the local RHS
+        noalias(rRightHandSideVector) += heat_flux_local * IntToReferenceWeight;
```

Comment:
if here u use += you must have set rRightHandSideVector to zero somewhere before

in any case, this code is not correct, you should do

     const auto& N = row(N_local, PointNumber ) //this gives you all the Ns associated to the gauss_point you are on
     const double heat_on_gauss = inner_prod(N, heat_flux_local) 
     noalias(rRightHandSideVector) += heat_on_gauss*N;